src/client/datascience/notebook/statusBarProvider.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { injectable } from 'inversify';
+import {
+    CancellationToken,
+    NotebookCell,
+    NotebookCellStatusBarAlignment,
+    NotebookCellStatusBarItem,
+    NotebookCellStatusBarItemProvider as VSCNotebookCellStatusBarItemProvider,
+    ProviderResult
+} from 'vscode';
+import { getCellStatusMessageBasedOnFirstCellErrorOutput } from './helpers/helpers';
+
+/**
+ * This class controls the status messages that we see on our Notebook cells
+ */
+@injectable()
+export class StatusBarProvider implements VSCNotebookCellStatusBarItemProvider {
+    // Weakmap, as we don't own the lifetime of either of these, just a helpful mapping for tests
+    private static cellStatusMappings = new WeakMap<NotebookCell, NotebookCellStatusBarItem>();
+
+    // Allow test code to see what status is currently on any cell
+    public static getCellStatusBarItem(cell: NotebookCell): NotebookCellStatusBarItem | undefined {
```

Comment:
Instead of adding to the existing tracker I just put the tracker for tests on there with the same basic scheme. Seemed easier than having to throw an event from here and handle it elsewhere since there is no notebook API for when cells get new status items.